### PR TITLE
mordernize cmake and update header paths for shared and getopts

### PR DESCRIPTION
**Pull request by ajakhotia**
This PR:
* Modernizes cmake and exports target metadata appropriately
* Moves files to facilitate above without changing the structure of the header except for 2 utility app(getopt & shared)
* Use built-in cmake feature to manage versioning and checking for compatibility on client call to find_package(MultiSense)

After these changes, a client may simply include this repository as a submodule or use find_package to find the targets from a install tree.

**Example 1:** Finding MultiSense using find_package. Cmake will automatically search all path listed in CMAKE_PREFIX_PATH and default system paths as setup by the client during the configure step of their project.
```
cmake_minimum_required(VERSION 3.10)
project(experiments)

find_package(MultiSense 3.7.0.0 REQUIRED)

add_executable(experiments main.cpp)
target_link_libraries(experiments MultiSense::MultiSense)
```

Example 2: Include MultiSense as a sub-repo with-in your own project's source tree.
```
add_subdirectory(LibMultiSense)

add_library(example SHARED ...)
target_link_libraries(example ... MultiSense::MultiSense)

# With the above, source of example should be able to find appropriate headers and link to build multisense library correctly.
```

### CODE REVIEW

**Comment by ajakhotia** [comment]
@QuentinTorg: Thanks for adding reviewers.
@mattalvarado: Can you please add me to the writers group for this repository?

**Comment by mattalvarado** [comment]
Sure I can give you access, but I would wait for a review before landing. 

Have you confirmed this will build under windows too?

**Comment by ajakhotia** [comment]
> Sure I can give you access, but I would wait for a review before landing.

Thank you. The PR is blocked from merge-into-master until I get approval.



> Have you confirmed this will build under windows too?
@QuentinTorg is doing the needed to verify the build under windows. I don't have a windows setup ready yet.

**Comment by mattalvarado** [comment]
@ajakhotia You should now have access. I will give this a review tonight

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,48 +1,69 @@
...
-#
+cmake_minimum_required(VERSION 3.5)
 
-#
-# Make sure there is a reasonable version of CMAKE installed.
-#
+project(LibMultiSense VERSION 3.9.0.0 LANGUAGES CXX C)
 
-cmake_minimum_required(VERSION 2.8)
+add_compile_options(-Wall -Wextra -Werror -Wpedantic)
```
Comment (mattalvarado) [inline CMakeLists.txt:5]: I know gcc 9.3.0 (default compiler on 20.04) has several issues with this library if `-Wpedantic` is enabled. We should probably fix those

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,48 +1,69 @@
...
 # Use full RPATH
 #
-set (CMAKE_SKIP_BUILD_RPATH FALSE)
-set (CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
-set (CMAKE_INSTALL_RPATH ${CMAKE_INSTALL_PREFIX}/lib)
-set (CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
+#set (CMAKE_SKIP_BUILD_RPATH FALSE)
+#set (CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
+#set (CMAKE_INSTALL_RPATH ${CMAKE_INSTALL_PREFIX}/lib)
+#set (CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
```
Comment (mattalvarado) [inline CMakeLists.txt:38]: These make it much more convenient for users who install LibMultiSense into `/usr/local`. I.e. `cmake -DCMAKE_INSTALL_PREFIX ../install ..` from the build directory will require users to add `../install/lib` to their LD_LIBRARY_PATH. You can confirm this by running any of the utilities in `<install directory>/bin`

File: LibMultiSense/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+target_link_libraries(MultiSense PRIVATE Threads::Threads)
+
+# TODO(ajakhotia): Evaluate if this is still necessary.
+#if(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
+#    target_link_libraries(MultiSense ws2_32)
+#elseif(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+#    target_link_libraries(MultiSense pthread)
+#else()
+#    target_link_libraries(MultiSense pthread rt)
+#endif()
```
Comment (mattalvarado) [inline LibMultiSense/CMakeLists.txt:47]: This needs to be evaluated before this lands

File: Utilities/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+#
+# else ()
+#
+# set (MULTISENSE_UTILITY_LIBS
+#     MultiSense
+#     pthread
+#     rt
+# )
+#
+# endif ()
```
Comment (mattalvarado) [inline Utilities/CMakeLists.txt:40]: This should be evaluated before this lands

**Pull request closed without merging**